pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -89,6 +89,59 @@ func (v *storagePolicyAPI) getDefaultDatastore(ctx context.Context) (*mo.Datasto
 	return &dsMo, nil
 }
 
+func (v *storagePolicyAPI) getDatastore(ctx context.Context, dcName string, dsName string) (*mo.Datastore, error) {
+	vmClient := v.vcenterApiConnection.Client
+	finder := find.NewFinder(vmClient.Client, false)
+
+	dc, err := finder.Datacenter(ctx, dcName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to access datacenter %s: %s", dcName, err)
+	}
+
+	finder = find.NewFinder(vmClient.Client, false)
+	finder.SetDatacenter(dc)
+	ds, err := finder.Datastore(ctx, dsName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to access datastore %s: %s", dsName, err)
+	}
+
+	var dsMo mo.Datastore
+	pc := property.DefaultCollector(dc.Client())
+	properties := []string{DatastoreInfoProperty, SummaryProperty}
+	err = pc.RetrieveOne(ctx, ds.Reference(), properties, &dsMo)
+	if err != nil {
+		return nil, fmt.Errorf("error getting properties of datastore %s: %v", dsName, err)
+	}
+	return &dsMo, nil
+}
+
+func (v *storagePolicyAPI) createZonalStoragePolicy(ctx context.Context) (string, error) {
+	var err error
+	failureDomains := v.infra.Spec.PlatformSpec.VSphere.FailureDomains
+	if len(failureDomains) <= 1 {
```

Comment:
Why would one failure domain be an error?

Reply:
Not a hard error as such, but given that this gets called only when `len(failureDomain) > 1`, it would be strange to have it less than one now.

Reply:
nit: the function name says it creates a zonal storage policy (singular), but the code creates multiple storage policies.

Reply:
No the code creates only one storage policy. It creates and attaches tags to multiple datastores found in each failure-domain.

Reply:
Sorry, I missed that